File: app/routers/branches.py
"""Branches router."""
from fastapi import APIRouter, Depends, HTTPException
from app.models.branch import BranchCreate, BranchUpdate
from app.models.common import ApiResponse
from app.repositories import branch_repo
from app.middleware.auth_middleware import require_role, get_current_user

router = APIRouter(prefix="/branches", tags=["Inventory"])


@router.get("", response_model=ApiResponse, summary="List branches")
def list_branches(user: dict = Depends(get_current_user)):
    items = branch_repo.get_all()
    return ApiResponse(ok=True, data=items)


@router.get("/{branch_id}", response_model=ApiResponse, summary="Get branch")
def get_branch(branch_id: int, user: dict = Depends(get_current_user)):
    b = branch_repo.get_by_id(branch_id)
    if not b:
        raise HTTPException(status_code=404, detail="Branch not found")
    return ApiResponse(ok=True, data=b)


@router.post("", response_model=ApiResponse, summary="Create branch")
def create_branch(payload: BranchCreate, user: dict = Depends(require_role(["owner"]))):
    branch_id = branch_repo.create(payload.code, payload.name)
    return ApiResponse(ok=True, data=branch_repo.get_by_id(branch_id))


@router.put("/{branch_id}", response_model=ApiResponse, summary="Update branch")
def update_branch(branch_id: int, payload: BranchUpdate, user: dict = Depends(require_role(["owner"]))):
    if not branch_repo.get_by_id(branch_id):
        raise HTTPException(status_code=404, detail="Branch not found")
    branch_repo.update(branch_id, payload.model_dump(exclude_unset=True))
    return ApiResponse(ok=True, data=branch_repo.get_by_id(branch_id))

File: frontend/src/app/api/branches.ts
import { apiRequest } from "./client";

export type BranchRow = {
  id: number;
  code: string;
  name: string;
  is_active?: boolean;
  created_at?: string | null;
};

export function getBranches(token: string) {
  return apiRequest<BranchRow[]>("/branches", { token });
}

export function createBranch(token: string, payload: { code: string; name: string }) {
  return apiRequest<BranchRow>("/branches", {
    method: "POST",
    token,
    body: payload
  });
}

export function updateBranch(
  token: string,
  branchId: number,
  payload: { name?: string; is_active?: boolean }
) {
  return apiRequest<BranchRow>(`/branches/${branchId}`, {
    method: "PUT",
    token,
    body: payload
  });
}
